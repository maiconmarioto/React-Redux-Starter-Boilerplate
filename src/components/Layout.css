  html {
    font-family: 'Roboto', sans-serif;
    background-color: lightgray;
  }

  .card {
    margin-top: 10vh;
    width: 400px;
  }

  .media {
    height: 0px;
    padding-top: 56.25%;
  }

  .center {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-content: center;

  }