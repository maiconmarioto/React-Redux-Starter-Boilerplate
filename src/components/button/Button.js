import React from "react";
import { Button as MaterialButton } from "@material-ui/core";

const Button = ({ children, onClick, disabled, variant, color }) => {
  return (
    <MaterialButton
      onClick={onClick}
      disabled={disabled}
      variant={variant}
      color={color}
    >
      {children}
    </MaterialButton>
  );
};

export default Button;
