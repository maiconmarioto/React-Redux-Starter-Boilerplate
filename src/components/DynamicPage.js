import React from "react";
import { Link } from "react-router-dom";

import Layout from "./Layout";

const DynamicPage = () => {
  return (
    <Layout>
      <h2>Dynamic Page</h2>
      <p>
        <Link to="/">Home</Link>
      </p>
      <p>This page was loaded asynchronously!!!</p>
    </Layout>
  );
};

export default DynamicPage;
