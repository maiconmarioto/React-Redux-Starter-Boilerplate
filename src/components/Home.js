import React from "react";
import { Link } from "react-router-dom";

import Layout from "./Layout";

const Home = () => {
  return (
    <Layout>
      <p>!!!</p>
      <p>
        <Link to="/dynamic">Dynamic Page</Link>
      </p>
    </Layout>
  );
};

export default Home;
