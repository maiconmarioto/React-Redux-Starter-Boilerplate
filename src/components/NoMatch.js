import React from "react";

import Layout from "./Layout";

const NoMatch = () => {
  return (
    <Layout>
      <strong>Page not found!</strong>
    </Layout>
  );
};

export default NoMatch;
