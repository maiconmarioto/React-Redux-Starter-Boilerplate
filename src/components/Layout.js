import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import CardMedia from "@material-ui/core/CardMedia";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";

import { card, media, center } from "./layout.css";

const Layout = ({ children }) => {
  return (
    <div className={center}>
      <Card className={card}>
        <CardMedia
          className={media}
          image="https://haxeflixel.com/documentation/images/00_getting_started/hello-world.png"
          title="Hello World!"
        />
        <CardContent>{children}</CardContent>
      </Card>
    </div>
  );
};

export default Layout;
