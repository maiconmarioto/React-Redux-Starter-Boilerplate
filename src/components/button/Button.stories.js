import React from "react";

import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";

import Button from "./Button";

storiesOf("Button", module)
  .add("default", () => (
    <Button onClick={action("clicked")}>Hello Button</Button>
  ))
  .add("disabled", () => (
    <Button onClick={action("clicked")} disabled>
      Hello Button
    </Button>
  ))
  .add("raised", () => (
    <Button color="primary" variant="raised" onClick={action("clicked")}>
      Hello Button
    </Button>
  ))
  .add("outlined", () => (
    <Button color="secondary" variant="outlined" onClick={action("clicked")}>
      Hello Button
    </Button>
  ));
