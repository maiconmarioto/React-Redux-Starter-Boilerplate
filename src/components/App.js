import React from "react";
import { Switch, BrowserRouter as Router, Route } from "react-router-dom";
import { hot } from "react-hot-loader";
import importedComponent from "react-imported-component";

import Home from "./Home";
import Loading from "./Loading";

const AsyncDynamicPAge = importedComponent(
  () => import(/* webpackChunkName:'DynamicPage' */ "./DynamicPage"),
  {
    LoadingComponent: Loading
  }
);
const AsyncNoMatch = importedComponent(
  () => import(/* webpackChunkName:'NoMatch' */ "./NoMatch"),
  {
    LoadingComponent: Loading
  }
);

const App = () => {
  return (
    <Router>
      <div>
        <Switch>
          <Route exact path="/" component={Home} />
          <Route exact path="/dynamic" component={AsyncDynamicPAge} />
          <Route component={AsyncNoMatch} />
        </Switch>
      </div>
    </Router>
  );
};

export default hot(module)(App);
