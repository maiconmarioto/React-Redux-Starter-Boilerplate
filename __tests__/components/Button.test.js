import React from "react";

import { Button } from "../../src/components/button";

const handleClick = () => {
  alert("clicked!");
};

describe(Button, () => {
  it("should render children text", () => {
    const button = shallow(<Button>Save</Button>);
    expect(button).toMatchSnapshot();
  });

  it("should handle click event", () => {
    const button = shallow(<Button onClick={handleClick} />);
    window.alert = jest.fn();
    button.simulate("click");
    expect(window.alert).toHaveBeenCalledWith("clicked!");
  });

  it("should render a raised button", () => {
    const wrapper = shallow(<Button variant="raised" />);
    expect(wrapper.prop("variant")).toEqual("raised");
  });
});
