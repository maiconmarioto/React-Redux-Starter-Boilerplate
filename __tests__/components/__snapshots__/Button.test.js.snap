// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render children text 1`] = `
<WithStyles(Button)>
  Save
</WithStyles(Button)>
`;
